README.md

```diff
@@ -1,1218 +1,533 @@
+
+## Networking
+
+```mermaid
+graph TD
+    A[Collector] -->|Send| B[Network Plugin]
+    B -->|Encrypt| C[Network]
+    C -->|Decrypt| D[Receiver]
+    D -->|Process| E[Storage]
+    
+    subgraph Security
+    F[Authentication]
+    G[Encryption]
+    end
+    
+    F -->|Verify| B
+    G -->|Protect| B
+```
+
+### Network Configuration
+
+```apache
+LoadPlugin network
+<Plugin network>
+  <Server "192.168.1.101" "25826">
+    SecurityLevel Encrypt
+    Username "user"
+    Password "secret"
+  </Server>
+</Plugin>
+```
+
+## Security
+
+### Security Features
+
+- Authentication with username/password
+- Network traffic encryption
+- Access control lists
+- Plugin-specific security options
+
+## Integration
+
+### Supported Integrations
+
+- **Visualization**
+  - Grafana
+  - Kibana
+  - Custom dashboards
+
+- **Storage Systems**
+  - Prometheus
+  - InfluxDB
+  - Elasticsearch
+  - OpenTSDB
+
+- **Processing Systems**
+  - Apache Kafka
+  - RabbitMQ
+  - Redis
+
+### Detailed Feature Breakdown
+
+#### Performance Metrics Collection
+- **System-level Metrics**
+  - CPU utilization per state (user, system, idle, etc.)
+  - Memory usage (free, used, buffered, cached)
+  - Disk I/O statistics (reads, writes, latency)
+  - Network interface throughput and errors
+  - Process statistics (CPU usage, memory usage, file descriptors)
+  - System load averages
+  - Temperature sensors
+  - Battery status
+
+- **Application-level Metrics**
+  - Database performance metrics (MySQL, PostgreSQL, MongoDB)
+  - Web server statistics (Apache, Nginx)
+  - Java application metrics through JMX
+  - Container metrics (Docker, LXC)
+  - Virtual machine statistics (libvirt)
+  - Message queue metrics (RabbitMQ, Apache Kafka)
+  - Cache server statistics (memcached, Redis)
+
+#### Advanced Data Processing
+- **Filtering Capabilities**
+  - Regular expression matching
+  - Threshold-based filtering
+  - Rate calculations
+  - Value aggregation
+  - Statistical analysis
+
+- **Notification System**
+  - Threshold-based alerts
+  - Trend analysis
+  - Custom notification scripts
+  - Integration with monitoring systems
+  - Email notifications
+  - SNMP traps
+
+[Previous Architecture section with added detail:]
+
+### Detailed Architecture Overview
+
+```mermaid
+graph TD
+    A[Input Sources] -->|Raw Data| B[Collection Layer]
+    B -->|Metrics| C[Processing Layer]
+    C -->|Processed Data| D[Output Layer]
+    
+    subgraph Input Sources
+    A1[System Metrics]
+    A2[Application Metrics]
+    A3[Custom Sources]
+    end
+    
+    subgraph Collection Layer
+    B1[Input Plugins]
+    B2[Plugin Registry]
+    B3[Collection Scheduler]
+    end
+    
+    subgraph Processing Layer
+    C1[Filter Chain]
+    C2[Aggregation]
+    C3[Threshold Checking]
+    end
+    
+    subgraph Output Layer
+    D1[Storage Plugins]
+    D2[Network Transport]
+    D3[Notification System]
+    end
+    
+    A1 & A2 & A3 --> B
+    B1 & B2 & B3 --> C
+    C1 & C2 & C3 --> D
+```
```

Comment:
IMHO redundant here and the resulting graph is too large.

Plugin section is missing a subsection on Processing plugins (aggregration / filtering), please add that instead.